Remove unused DynamicDAG class.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24890 Remove unused DynamicDAG class.**

Differential Revision: [D16912935](https://our.internmc.facebook.com/intern/diff/D16912935)

Commit:
Remove unused DynamicDAG class.

ZolotukhinM commented:
@suo @zdevito do you know what DynamicDAG is for? I couldn't find any of its uses and I think it might be a leftover of some unfinished effort. Is it ok to remove it?

suo commented:
It was the predecessor to the topological mover stuff. It's safe to remove. Thanks!

Commit:
Rebase on "Remove unused DynamicDAG class."


Differential Revision: [D16912935](https://our.internmc.facebook.com/intern/diff/D16912935)
